lib/ts/recipe/session/with-jwt/utils.ts

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { NormalisedAppinfo } from "../../../types";
+import { ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY } from "./constants";
+import { RecipeInterface as JWTRecipeInterface } from "../../jwt/types";
+
+export async function addJWTToAccessTokenPayload({
+    accessTokenPayload,
+    jwtExpiry,
+    userId,
+    jwtPropertyName,
+    appInfo,
+    jwtRecipeImplementation,
+}: {
+    accessTokenPayload: any;
+    jwtExpiry: number;
+    userId: string;
+    jwtPropertyName: string;
+    appInfo: NormalisedAppinfo;
+    jwtRecipeImplementation: JWTRecipeInterface;
+}): Promise<any> {
+    // If jwtPropertyName is not undefined it means that the JWT was added to the access token payload already
+    let existingJwtPropertyName = accessTokenPayload[ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY];
+
+    if (existingJwtPropertyName !== undefined) {
+        // Delete the old JWT and the old property name
+        delete accessTokenPayload[existingJwtPropertyName];
+        delete accessTokenPayload[ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY];
+    }
+
+    // Update the access token payload with the default claims to add
+    accessTokenPayload = {
+        /* 
+            We add our claims before the user provided ones so that if they use the same claims
+            then the final payload will use the values they provide
+        */
+        sub: userId,
+        iss: appInfo.apiDomain.getAsStringDangerous(),
+        ...accessTokenPayload,
+    };
+
+    // Create the JWT
+    let jwtResponse = await jwtRecipeImplementation.createJWT({
+        payload: accessTokenPayload,
+        validitySeconds: jwtExpiry,
+    });
```

Comment:
So we should not add the `sub` and `iss` to the actual `accessTokenPayload`, but we should only add it to the `payload` param when calling `createJWT`. This way, we save some space in the overall cookie size.

Reply:
Done